### DISCUSSION TITLE
hidden pages show up in navigation when acitve

### DISCUSSION THREAD

**Discussion by andi-blafasl** [McShelby/hugo-theme-relearn#138] [Q&A] [answered]
Not sure if this a Bug from the Theme or a Problem with my content structure.

I have hidden pages and, as expected, they don't show up in the navigation. They are linked manually under the "More" section. When they are shown, they appear in the navigation above the "more" section.

I hope you understand my description. You can see and test the behavior on my Site [BITS-Training.de](https://www.bits-training.de/) and the Source for Hugo is available as a Git Repository on GitHub [BITS-Hugo](https://github.com/BITS-Training/BITS-hugo)

**Comment by McShelby** [comment, marked as answer]
So at first: This works as designed.

Detailed explaination: If a page is hidden but resides somewhere inside your page structure, you want to see it at that certain position in the menu if the hidden page is shown. This is the same case with page-1-1-1-1-1 in the exampleSite.

A workaround would be to place your shortcut-pages in your sites root. Similar to the credits.html in the exampleSite. In this case you can remove the `hidden` frontmatter parameter.

**Comment by andi-blafasl** [comment]
Just tested your workaround and it is looking good. We will make the change in the next release. Thank you for the good explanation. Maybe you can note this somewhere in the documentation. Maybe this is helpful for other users in the future.

**Comment by McShelby** [comment]
For now, I'll move this to discussions and keep your remark in mind.

Improving documentation is an ongoing issue. I for myself am not happy with the current structuring. So you'll may see some improvements in the near future.

**Comment by cagix** [comment]
@andi-blafasl @McShelby I have a related issue: I have some pages that are linked directly in the "More" section and should not appear in the menu. 

You can set `hidden: true` in the YAML header of these pages. This will prevent these pages from appearing in the menu. However, as soon as you open that page, it will then appear in the menu. 

I don't remember where I found it, but if you put the snippet

```
_build:
  render: always
  list: never
  publishResources: true
```

in the YAML header, these pages remain hidden even if you visit them :)  It seems sufficient if this snippet is included in the YAML of the respective `_index.md`.

**Reply by McShelby** [reply]
In Hugo terms this is called a headless bundle and ist the recommended way top add such pages. You can see this in the exampleSite of this theme, too.

**Reply by cagix** [reply]
Interestingly according to https://gohugo.io/content-management/page-bundles/#headless-bundle a simple `headless: true` in the `index.md` should do the trick. However, that seems not to work in my case, i.e. I still need to add the `_build` section to my YAML. Hmmm.
